src/types/message.ts

```diff
@@ -13,8 +13,15 @@ export enum ContentTypeTagName {
   toolCall = 'tool-call',
 }
 
+export type PromptlSourceRef = {
+  start: number
+  end: number
+  identifier?: string
+}
+
 interface IMessageContent {
   type: ContentType
+  _promptl_source_map?: PromptlSourceRef[]
```

Comment:
Actually, better use `_promptlSourceMap` as we're using camelCase as standard

Reply:
Even better namespace with latitude `_latitudePromptSourceMap`